route/build/src/read_restart.f90

```diff
@@ -131,6 +136,11 @@ SUBROUTINE read_state_nc(fname,           &   ! input:  state netcdf name
    if(ierr/=0)then; message=trim(message)//trim(cmessage);return; endif
  end if
 
+ if (tracer) then
+   call read_solute_state(idxDW, ierr, message)
+   if(ierr/=0)then; message=trim(message)//trim(cmessage);return; endif
+ end if
+
  CONTAINS
 
```

Comment:
Add a description for all the read_*_state subroutines. To describe what they are all doing. And that they should all be identical other than the type that they read.

So for example something like:

``` fortran
! *********************************************************************
! private subroutines: read the state for each specific variable type
! Description: There is a separate subroutine for each variable type, for example: basinQ, bas_solute, IRF, etc. 
! They all work the same in how they read in the state data, but just operate on a different type.
! *********************************************************************

SUBROUTINE read_basinQ_state(ierr, message1)
.
.
.

SUBROUTINE read_bas_solute_state(ierr, message1)
```